test/e2e_node/oomkiller_test.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+)
+
+type testCase struct {
+	podSpec                *v1.Pod
+	oomTargetContainerName string
+}
+
+const PodOOMKilledTimeout = 2 * time.Minute
+
+var _ = SIGDescribe("OOMKiller [LinuxOnly] [NodeConformance]", func() {
```

Comment:
I thought that the OOMKiller follows some heuristics, is this test guarantee to kill ONLY the pod under test?
should we tag it as Disruptive if not?

Reply:
The intention of the test is to induce oom killer by the pod exceeding its limits rather than putting the system low on memory (the pod consumes 20mb which should not put the machine under memory stress).  

However, I'm not sure about guarantees and I'm ok with the `Disruptive` label. Maybe it is just safer to add it? @bobbypage  @haircommander ?

Reply:
ah, nevermind, scratch my comment, is about cgroups oomkiller,